### PR TITLE
Convert markup format of source codes from RDoc to Markdown

### PR DESCRIPTION
**Pull request by kenichiice**
see #163 

This PR makes following changes. These should improve the look of the reference manual.

* Change markup format of source codes from RDoc to Markdown.
    * Convert "heading level 1": `=` --> `#`
    * Convert "heading level 2": `==` --> `##`
    * Convert "code": `+WORD+` --> <code>\`WORD\`</code>
    * Convert "definition lists".
        * However, "Author", "Copyright" and "License" at the beginning of the source code have not changed.
    * Change indents of "code block".
    * Insert necessary blank lines.
* Change `ja.po`

### CODE REVIEW

File: lib/test/unit/notification.rb [hunk 1]
```diff
@@ -65,15 +65,17 @@ def included(base)
...
-      #     notify("Special!") if special_case?
-      #     # Reached here too
-      #   end
+      #
+      #     def test_notification
+      #       notify("I'm here!")
+      #       # Reached here
+      #       notify("Special!") if special_case?
+      #       # Reached here too
+      #     end
```
Comment (kou) [inline lib/test/unit/notification.rb]: We can use `@example` tag for YARD: https://www.rubydoc.info/gems/yard/file/docs/Tags.md#example
But this is out-of-scope of this pull request. So we can this as is in this pull request.

File: lib/test/unit/notification.rb [hunk 2]
```diff
@@ -65,15 +65,17 @@ def included(base)
...
       #
       # options:
-      #   :backtrace override backtrace.
+      #
+      #     :backtrace override backtrace.
```
Comment (kou) [inline lib/test/unit/notification.rb]: This is not pre-formatted text. But we can work on this as follow-up pull request.

**Comment by hiroyuki-sato** [comment]
Hello, @kenichiice 
Thank you for creating this PR. 
I updated [test-unit document](https://test-unit.github.io/test-unit/en/Test/Unit/Assertions.html)
Please take a look when you get a chance.

test-unit

**Comment by kenichiice** [comment]
Thank you. It looks as expected.

**Pull request merged**